docs/design/2018-10-24-view-support.md

```diff
@@ -0,0 +1,69 @@
+
+Below is a visual depiction of a view:
+  ![AnatomyOfAview](imgs/view.png)
+  
+A view is created from a query using the CREATE OR REPLACE VIEW command. In the example below we are creating a PopularBooks view based on a query which selects all Books that have the IsPopular field checked. Following is the query:
+`CREATE OR REPLACE VIEW PopularBooks AS SELECT ISBN, Title, Author,PublishDate FROM Books WHERE IsPopular=1`  
+
+Once a view is created, you can use it as you query any table in a SELECT statement. For example, to list all the popular book titles ordered by an author, you could write:  
+`SELECT Author, Title FROM PopularBooks ORDER BY Author`  
+
+In general you can use any of the SELECT clauses, such as GROUP BY, in a select statement containing a view.
+
+## Proposal
+I implement the `CREATE OR REPLACE` command to create a view with the SELECT statement. Currently, I reuse `DROP TABLE` to drop a view, and `show full tables` can also display the view list with different table types. The `show create table` command can also regenerate a view create command.
+
+## Rationale
+View is just a tableinfo object store with select statement, view only support two types ddl operation, which are `CREATE` and `DROP`. Currently, view can only used within `SELECT` statement.  
+Let me describe some more detail about view ddl operation:
+1. Create a VIEW  
+  1.1 Parse the create view statement and build a logical plan for select cause part, if any grammar error occur, return errors to parser.   
+  1.2 Examine view definer's privileges, definer should own both `CREATE_VIEW` and base table's `SELECT` privileges.  
+  1.3 If `ViewFieldList` cause is empty, then we should generate view column name from `SelectStmt` cause.  
+  1.4 Replace any wildcard with column name in `SELECT` cause.  
+  1.5 Test `SELECT` cause with some rules to see if this view is updateable.  
+  1.6 Store the view definition into following tables
+     * `INFORMATION_SCHEMA.VIEW_COLUMN_USAGE`  Lists one row for each column in a view including the base table of the column where possible 
+     * `INFORMATION_SCHEMA.VIEW_TABLE_USAGE`  Lists one row for each table used in a view 
+     * `INFORMATION_SCHEMA.VIEWS` Lists one row for each view
+      
+2. Drop a view  
+  Implement DROP VIEW grammer, delete exist view tableinfo object.  
+3. Select from a view  
+  3.1 In function `func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error)`, if `tn *ast.TableName` is a view, then we build a select logicalplan from view's view_select string.   
+4. Insert/Update into a view  
+  I do not have any idea now, will implement later.   
+5. Describe a view  
+  Generate the view select logicalplan, if any error occur return error to parser. Otherwise return `SELECT` cause's column infos.
```

Comment:
```suggestion
  Generate the view select logicalplan. If any error occurs, return the error to parser. Otherwise, return `SELECT` cause's column info.
```